es/ponyfill.js

```diff
@@ -1,17 +1,23 @@
+
+const observableSymbols = new WeakMap();
+
 export default function symbolObservablePonyfill(root) {
-	var result;
-	var Symbol = root.Symbol;
+	var result = observableSymbols.get(root);
+	if (result) {
+		return result;
+	}
 
+	var Symbol = root.Symbol;
 	if (typeof Symbol === 'function') {
 		if (Symbol.observable) {
 			result = Symbol.observable;
 		} else {
 			result = Symbol('observable');
-			Symbol.observable = result;
```

Comment:
Again, I'm not sure if I understand the goal of removing the polyfilling from this?

Is it that it causes an error in some runtimes?

Reply:
Yes. With the XS runtime and with SES, which both protect client code from prototype pollution, Symbol is a frozen object. My understanding is that the useful distinction between a ponyfill and a polyfill is that the former is useful in such environments and doesn’t require the special treatment of shims.